kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala

```diff
@@ -58,14 +61,30 @@ class FlinkProcessBuilder(
   // flink.execution.target are required in Kyuubi conf currently
   val executionTarget: Option[String] = conf.getOption("flink.execution.target")
 
+  private lazy val proxyUserEnable: Boolean = {
+    doAsEnabled && conf.get(ENGINE_FLINK_DOAS_ENABLED) &&
+    conf.getOption(s"$FLINK_CONF_PREFIX.$FLINK_SECURITY_KEYTAB_KEY").isEmpty &&
+    conf.getOption(s"$FLINK_CONF_PREFIX.$FLINK_SECURITY_PRINCIPAL_KEY").isEmpty
+  }
+
   override protected def module: String = "kyuubi-flink-sql-engine"
 
   override protected def mainClass: String = "org.apache.kyuubi.engine.flink.FlinkSQLEngine"
 
-  override def env: Map[String, String] = conf.getEnvs +
-    ("FLINK_CONF_DIR" -> conf.getEnvs.getOrElse(
-      "FLINK_CONF_DIR",
-      s"$flinkHome${File.separator}conf"))
+  override def env: Map[String, String] = {
+    val flinkExtraEnvs = if (proxyUserEnable) {
+      Map(
+        "FLINK_CONF_DIR" -> conf.getEnvs.getOrElse(
+          "FLINK_CONF_DIR",
+          s"$flinkHome${File.separator}conf"),
```

Comment:
code style here is a little bit ugly. does int work on Windows? if not, just replace `${File.separator}` with `/`

Reply:
> does int work on Windows?

I'm not sure, but there are many places that use it.

![image](https://github.com/apache/kyuubi/assets/17894939/aeeac49d-39c0-428e-a477-ab2f23a67a37)

Reply:
how about extracting `FLINK_CONF_DIR` as a variable?
```
    val FLINK_CONF_DIR =
      conf.getEnvs.getOrElse("FLINK_CONF_DIR", s"$FLINK_HOME${File.separator}conf")
```

Reply:
thanks, changed